apps/studio/components/interfaces/Settings/API/ServiceList.tsx

```diff
@@ -28,6 +32,12 @@ export const ServiceList = () => {
   } = useReadReplicasQuery({ projectRef })
   const { data: loadBalancers } = useLoadBalancersQuery({ projectRef })
 
+  useEffect(() => {
+    if (querySource && querySource !== state.selectedDatabaseId) {
+      state.setSelectedDatabaseId(querySource)
+    }
+  }, [querySource, state, projectRef])
```

Comment:
There wasn't anything setting this from the URL param, so I added this logic to parse and read